project/Build.scala

```diff
@@ -1005,6 +1004,34 @@ object Build {
       ) ++ (
           scalaJSExternalCompileSettings
       ) ++ inConfig(Compile)(Seq(
+          scalacOptions in doc ++= Seq("-implicits", "-groups"),
+
+          // Filter doc sources to remove implementation details from doc.
+          sources in doc := {
+            def regexFileFilter(s: String): FileFilter = new FileFilter {
+              private val pat = s.r.pattern
+              override def accept(f: File): Boolean = {
+                val path = f.getAbsolutePath.replace('\\', '/')
+                pat.matcher(path).matches()
+              }
+            }
+
+            val filter: FileFilter = (
+                AllPassFilter
+                  -- regexFileFilter(".*/scala/scalajs/runtime/.*")
+                  -- regexFileFilter(".*/scala/scalajs/js/annotation/internal/.*")
+                  -- "*.nodoc.scala"
```

Comment:
Given the two uses of `regexFileFilter`, a regex is a bit overkill. You could have a `containsFileFilter("/scala/scalajs/runtime/")` that simply performs `path.contains(s)`.